cyprecice/cyprecice.pyx

```diff
@@ -699,186 +776,55 @@ cdef class Interface:
-
-        cdef bytes mesh_name_py_bytes = mesh_name.encode()
-        cdef bytes data_name_py_bytes = data_name.encode()
-
-        self.thisptr.writeVectorData (<const char*> mesh_name_py_bytes, <const char*> data_name_py_bytes, vertex_id, <const double*>_value.data)
-
-    def write_block_scalar_data (self, mesh_name, data_name, vertex_ids, values):
-        """
-        Writes scalar data given as a block. This function writes values of specified vertices to a dataID.
-
-        Parameters
-        ----------
-        mesh_name : str
-            Name of the mesh to write to.
-        data_name : str
-            ID to write to.
-        vertex_ids : array_like
-            Indices of the vertices.
-        values : array_like
-            Values to be written
-
-        Notes
-        -----
-        Previous calls:
-            count of available elements at values matches the given size
-            count of available elements at vertex_ids matches the given size
-            initialize() has been called
-
-        Examples
-        --------
-        Write block scalar data for a 2D and 3D problem with 5 (N=5) vertices:
-        >>> mesh_name = "MeshOne"
-        >>> data_name = "DataOne"
-        >>> vertex_ids = [1, 2, 3, 4, 5]
-        >>> values = np.array([v1, v2, v3, v4, v5])
-        >>> interface.write_block_scalar_data(mesh_name, data_name, vertex_ids, values)
-        """
-        check_array_like(vertex_ids, "vertex_ids", "write_block_scalar_data")
-        check_array_like(values, "values", "write_block_scalar_data")
-
-        if len(values) > 0:
-            assert(len(vertex_ids) == len(values))
-            size = len(vertex_ids)
-        if len(values) == 0:
-            size = 0
+            assert dimensions == self.get_data_dimensions(mesh_name, data_name), "Dimensions of vector data in write_data do not match with dimensions in problem definition. Provided dimensions: {}, expected dimensions: {}".format(dimensions, self.get_data_dimensions(mesh_name, data_name))
 
-        cdef np.ndarray[int, ndim=1] _vertex_ids = np.ascontiguousarray(vertex_ids, dtype=np.int32)
-        cdef np.ndarray[double, ndim=1] _values = np.ascontiguousarray(values, dtype=np.double)
+        assert len(vertex_ids) == size, "Vertex IDs are of incorrect length in write_data. Check length of vertex ids input. Provided size: {}, expected size: {}".format(vertex_ids.size, size)
 
-        assert _values.size == size, "Scalar data is not provided for all vertices in write_block_scalar_data. Check size of input data provided. Provided size: {}, expected size: {}".format(_values.size, size)
-        assert _vertex_ids.size == size, "Vertex IDs are of incorrect length in write_block_scalar_data. Check size of vertex ids input. Provided size: {}, expected size: {}".format(_vertex_ids.size, size)
+        cdef vector[int] cpp_ids = vertex_ids
+        cdef vector[double] cpp_values = values.flatten()
```

Comment:
Would be good, if we could avoid this explicit copying of data and use ranges instead. But let's do this as a follow-up.